shim/shim.c

```diff
@@ -296,25 +297,67 @@ handle_data(int infd, int outfd)
 	}
 }
 
+/*!
+ * Print help usage
+ *
+ * \param prog Binary name
+ */
+void print_usage(char *prog) {
+	printf("%s: Usage\n", prog);
+        printf("  -c,  --container_id   Container id\n");
+        printf("  -p,  --proxy_sock_fd  File descriptor of the socket connected to cc-proxy\n");
+        printf("  -h,  --help           Display this help message\n");
+}
+
 int
 main(int argc, char **argv)
 {
 	int proxy_sockfd;
 	struct sockaddr_un proxy_address;
-	char* container_id;
+	char* container_id = NULL;
 	char* proxy_sock_path;
 	struct pollfd poll_fds[MAX_POLL_FDS];
 	nfds_t nfds = 0;
 	int ret;
 	struct sigaction sa;
+	int c;
+	int proxy_sock_fd = -1;
+	char *endptr;
+
+	struct option prog_opts[] = {
+		{"container_id", required_argument, 0, 'c'},
+		{"proxy_sock_fd", required_argument, 0, 'p'},
+		{"help", no_argument, 0, 'h'},
+		{ 0, 0, 0, 0},
+	};
+
+	while ((c = getopt_long(argc, argv, "c:p:h", prog_opts, NULL))!= -1) {
+		switch (c) {
+			case 'c':
+				container_id = strdup(optarg);
+				break;
+			case 'p':
+				if ((ret = (int)strtol(optarg, &endptr, 10)) == -1 || *endptr != '\0') {
+					err_exit("Invalid value for proxy socket fd\n");
+				}
+				proxy_sock_fd = ret;
+				break;
+			case 'h':
+				print_usage(argv[0]);
+				exit(0);
+			default:
+				print_usage(argv[0]);
+				exit(EXIT_FAILURE);
+		}
+	}
 
-	if (argc != 3) {
-		printf("Usage: cc-shim container_id proxy_socket_path");
-		exit(EXIT_FAILURE);
+	if ( !container_id) {
+		err_exit("cc-shim: Missing container id\n");
```

Comment:
Hard-coding program name - not major, but `print_usage()` is passed the program name, so ideally `err_exit()` would do something similar (or maybe rely on a global `static const char *program_name` set in `main()`.